File: web/src/models/Post.ts
/**
 * Post.ts
 * DailySNSWithYourSS Web版
 * 
 * 投稿とアイテムのデータモデル
 */

export enum PostItemType {
  Screenshot = 'screenshot',
  Comment = 'comment'
}

export interface PostItem {
  id: string;
  type: PostItemType;
  content: string; // Image URL for screenshot, text for comment
  timestamp: Date;
  isDeleted: boolean;
}

export interface Post {
  id: string;
  userId: string;
  items: PostItem[];
  createdAt: Date;
  updatedAt: Date;
}

/**
 * PostItemを作成
 */
export function createPostItem(
  type: PostItemType,
  content: string,
  id?: string,
  timestamp?: Date
): PostItem {
  return {
    id: id || crypto.randomUUID(),
    type,
    content,
    timestamp: timestamp || new Date(),
    isDeleted: false
  };
}

/**
 * Postを作成
 */
export function createPost(
  userId: string,
  items: PostItem[] = [],
  id?: string,
  createdAt?: Date,
  updatedAt?: Date
): Post {
  const now = new Date();
  return {
    id: id || crypto.randomUUID(),
    userId,
    items,
    createdAt: createdAt || now,
    updatedAt: updatedAt || now
  };
}

/**
 * アクティブなアイテム（削除されていない）を取得
 */
export function getActiveItems(post: Post): PostItem[] {
  return post.items.filter(item => !item.isDeleted);
}

File: Sources/DailySNSWithYourSS/Post.swift
//
//  Post.swift
//  DailySNSWithYourSS
//
//  Created on 2025-11-23.
//

import Foundation

/// Represents a single post item (screenshot or comment)
public struct PostItem: Identifiable, Codable, Sendable {
    public let id: UUID
    public let type: PostItemType
    public let content: String // Image URL for screenshot, text for comment
    public let timestamp: Date
    public var isDeleted: Bool = false
    
    public enum PostItemType: String, Codable, Sendable {
        case screenshot
        case comment
    }
    
    public init(id: UUID = UUID(), type: PostItemType, content: String, timestamp: Date = Date()) {
        self.id = id
        self.type = type
        self.content = content
        self.timestamp = timestamp
    }
}

/// Represents a complete post with multiple items (screenshots and comments)
public struct Post: Identifiable, Codable, Sendable {
    public let id: UUID
    public let userId: String
    public var items: [PostItem]
    public let createdAt: Date
    public var updatedAt: Date
    
    public init(id: UUID = UUID(), userId: String, items: [PostItem] = [], createdAt: Date = Date(), updatedAt: Date = Date()) {
        self.id = id
        self.userId = userId
        self.items = items
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
    
    /// Get all non-deleted items
    public var activeItems: [PostItem] {
        items.filter { !$0.isDeleted }
    }
}
